### PR TITLE
Django 4 compatibility

### PR DESCRIPTION
**Pull request by Siburg**
Builds on earlier PR #337 from @helmetwearer to provide compatibility with Django 4.x.

Also extends custom `save` methods on models to ensure that `update_fields` include any fields updated in those methods, if necessary ( https://docs.djangoproject.com/en/4.2/releases/4.2/#setting-update-fields-in-model-save-may-now-be-required). That may be regarded as a case of YAGNI, but should avoid errors for any future user that may use an `update_or_create` query on those models.

And responded to #199 by replacing `admin_static` with `static` tag in admin template. I don't know if that will be problematic for Django versions earlier than 2.1, but considering that those are well out of date I expect the risk is low.

### CODE REVIEW

File: chroniker/admin.py [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
 from django.db import models
 from django.forms import TextInput
 from django.shortcuts import render
-from django.utils.encoding import force_text
+from django.urls import reverse, NoReverseMatch
+try:
+    from django.urls import re_path
+except ImportError:
+    from django.conf.urls import url as re_path
+from django.utils.encoding import force_str
```
Comment (Siburg) [inline chroniker/admin.py:13]: `force_str` was added in Django 1.5, so it seems unlikely that we still need a backward compatible import for `force_text`.

Comment (helmetwearer) [reply chroniker/admin.py:13]: Is the community off Python 2 yet? Might be worth polling but not sure how

**Comment by helmetwearer** [comment]
When trying to install in a docker/virtualenv setup (django 4.2 is installed)
pip install git+https://github.com/Siburg/django-chroniker.git@django_4_compatibility
I get:
File "/tmp/pip-req-build-xo1d0ymg/chroniker/__init__.py", line 1, in <module>
          from django import VERSION as DJANGO_VERSION
      ModuleNotFoundError: No module named 'django'
      
I'm not sure what the difference is

**Comment by Siburg** [comment]
>I get: File "/tmp/pip-req-build-xo1d0ymg/chroniker/**init**.py", line 1, in from django import VERSION as DJANGO_VERSION ModuleNotFoundError: No module named 'django'
> 

Strange. I just tried installing my branch into the dev environment (using `venv`) for my own project, which uses Django 4.2. I don't get the error you  mention, but I get 
```
Error fetching command 'calculate_job_chain': No module named 'criticalpath'
Command 'calculate_job_chain' skipped
```

Will need to look into that more.

**Comment by chrisspen** [comment]
Sorry for the late reply. I've fixed some things upstream as well as enabled a Github CI build. Would you mind merging with master and resolving the conflicts so the test suite can run?

File: chroniker/tests/urls.py [hunk 2]
```diff
@@ -16,17 +10,22 @@
...
 try:
     _patterns = [
-        url(r'^admin/', include(admin.site.urls)),
+        re_path(r'^admin/', include(admin.site.urls)),
     ]
 except ImproperlyConfigured:
     # Django >= 2.1.7.
     _patterns = [
-        url(r'^admin/', admin.site.urls),
+        re_path(r'^admin/', admin.site.urls),
```
Comment (Siburg) [inline chroniker/tests/urls.py:28]: For consisteny with admin, should maybe have left these as `url` as well. Or, persisted with my earlier change from `url` to `re_path` in the admin as well (which I reverted).

**Comment by chrisspen** [comment]
Thanks. It didn't look like your changes triggered a build, so I uploaded your fork myself and made updates for a few errors I ran into.

https://github.com/chrisspen/django-chroniker/actions/runs/9686092862/job/26727724787?pr=369

The Django 3.2 still passes, but there's something in Django 4 that's breaking how django-admin.py detects packages and it can't even find the chroniker package. I'll continue to look into it, but if you see what's causing that, please let me know.

**Comment by chrisspen** [comment]
For those who are curious, the issue was that Django renamed django-admin.py to django-admin, and for some odd reason the build wasn't returning any helpful error message like "django-admin.py not found".

Thanks for your help on the upgrade.

**Comment by Siburg** [comment]
> For those who are curious, the issue was that Django renamed django-admin.py to django-admin, and for some odd reason the build wasn't returning any helpful error message like "django-admin.py not found".
> 
Glad to hear you found it. Sorry for my late response. I was planning to look into it over the weekend, but happy that's no longer necessary.

**Pull request merged**